include/eve/module/core/regular/impl/simd/x86/gather.hpp

```diff
@@ -0,0 +1,162 @@
+    constexpr bool i_64x4   = match(i, int64x4 , uint64x4);
+    constexpr bool i_64x8   = match(i, int64x8 , uint64x8);
+
+    constexpr bool d_i32x4  = match(c, int32x4 , uint32x4);
+    constexpr bool d_i32x8  = match(c, int32x8 , uint32x8);
+    constexpr bool d_i32x16 = match(c, int32x16, uint32x16);
+    constexpr bool d_i64x2  = match(c, int64x2 , uint64x2);
+    constexpr bool d_i64x4  = match(c, int64x4 , uint64x4);
+    constexpr bool d_i64x8  = match(c, int64x8 , uint64x8);
+
+    // AVX512 note: yes, it's (v,p) and not (p,v), don't ask...
+    // 64 bits integer data
+    if      constexpr(i_32x8 && d_i64x8)          return _mm512_i32gather_epi64(v       , p, 8);
+    else if constexpr(i_32x4 && d_i64x2)          return _mm_i32gather_epi64   (pl_t(p) , v, 8);
+    else if constexpr(i_32x4 && d_i64x4)          return _mm256_i32gather_epi64(pl_t(p) , v, 8);
+    else if constexpr(i_64x8 && d_i64x8)          return _mm512_i64gather_epi64(v       , p, 8);
+    else if constexpr(i_64x4 && d_i64x4)          return _mm256_i64gather_epi64(pl_t(p) , v, 8);
+    else if constexpr(i_64x2 && d_i64x2)          return _mm_i64gather_epi64   (pl_t(p) , v, 8);
+    // 64 bits float data
+    else if constexpr(i_32x8 && c == float64x8)   return _mm512_i32gather_pd(v, p, 8);
+    else if constexpr(i_32x4 && c == float64x4)   return _mm256_i32gather_pd(p, v, 8);
+    else if constexpr(i_32x4 && c == float64x2)   return _mm_i32gather_pd   (p, v, 8);
+    else if constexpr(i_64x8 && c == float64x8)   return _mm512_i64gather_pd(v, p, 8);
+    else if constexpr(i_64x4 && c == float64x4)   return _mm256_i64gather_pd(p, v, 8);
+    else if constexpr(i_64x2 && c == float64x2)   return _mm_i64gather_pd   (p, v, 8);
+    // 32 bits integer data
+    else if constexpr(i_32x16 && d_i32x16)        return _mm512_i32gather_epi32(v,p,4);
+    else if constexpr(i_32x8  && d_i32x8 )        return _mm256_i32gather_epi32(pi_t(p),v,4);
+    else if constexpr(i_32x4  && d_i32x4 )        return _mm_i32gather_epi32   (pi_t(p),v,4);
+    else if constexpr(i_64x8  && d_i32x8 )        return _mm512_i64gather_epi32(v,p,4);
+    else if constexpr(i_64x4  && d_i32x4 )        return _mm256_i64gather_epi32(pi_t(p),v,4);
+    else if constexpr(i_64x2  && d_i32x4 )        return _mm_i64gather_epi32   (pi_t(p),v,4);
+    // 32 bits float data
+    else if constexpr(i_32x16 && c == float32x16) return _mm512_i32gather_ps(v, p, 4);
+    else if constexpr(i_32x8  && c == float32x8 ) return _mm256_i32gather_ps(p, v, 4);
+    else if constexpr(i_32x4  && c == float32x4 ) return _mm_i32gather_ps   (p, v, 4);
+    else if constexpr(i_64x8  && c == float32x8 ) return _mm512_i64gather_ps(v, p, 4);
+    else if constexpr(i_64x4  && c == float32x4 ) return _mm256_i64gather_ps(p, v, 4);
+    else if constexpr(i_64x2  && c == float32x4 ) return _mm_i64gather_ps   (p, v, 4);
+    // Aggregation cases
+    else if constexpr(has_aggregated_abi_v<wide<U, N>>)
```

Comment:
Can you move this up? Also I'd prefer the condition on U to be explicit tbh.

Reply:
OK for moving up.
Which condition on U, the index typeing instead of using the i??x? notation ?

Reply:
Not instead, just:

`sizeof(U) <= 2U` - go generic or smth

Reply:
Done